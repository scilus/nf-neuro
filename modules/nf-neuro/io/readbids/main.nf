process IO_READBIDS {
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://scil.usherbrooke.ca/containers/scilus_2.0.2.sif':
        'scilus/scilus:2.0.2' }"

    input:
        tuple path(bids_folder), path(fs_folder), path(bids_ignore)

    output:
        path("tractoflow_bids_struct.json")             , emit: bidsstructure
        path "versions.yml"                             , emit: versions


    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def fs_folder = fs_folder ? "--fs $fs_folder" : ''
    def bids_ignore = bids_ignore ? "--bids_ignore $bids_ignore" : ''
    def readout = task.ext.readout ? "--readout " + task.ext.readout : ''
    def clean_flag = task.ext.clean_bids ? "--clean " : ''

    """
    scil_bids_validate.py $bids_folder tractoflow_bids_struct.json\
        $readout \
        $clean_flag\
        $fs_folder\
        $bids_ignore\
        -v

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(pip list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """


    stub:
    def args = task.ext.args ?: ''
    """
    scil_bids_validate.py -h

    touch tractoflow_bids_struct.json

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(pip list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """
}
