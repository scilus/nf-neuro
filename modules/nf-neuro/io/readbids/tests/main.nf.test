nextflow_process {

    name "Test Process IO_READBIDS"
    script "../main.nf"
    process "IO_READBIDS"

    tag "modules"
    tag "modules_nfcore"
    tag "io"
    tag "io/readbids"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    test("io - readbids - folder") {

        setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "bids.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        file("\${test_data_directory}/i_bids", checkIfExists: true),
                        [],
                        [],
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
