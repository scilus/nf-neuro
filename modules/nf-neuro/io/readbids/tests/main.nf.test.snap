{
    "io - readbids - folder": {
        "content": [
            {
                "0": [
                    "tractoflow_bids_struct.json:md5,58e0494c51d30eb3494f7c9198986bb9"
                ],
                "1": [
                    "versions.yml:md5,53cf301d7d6076acbc0691d3e7e52437"
                ],
                "bidsstructure": [
                    "tractoflow_bids_struct.json:md5,58e0494c51d30eb3494f7c9198986bb9"
                ],
                "versions": [
                    "versions.yml:md5,53cf301d7d6076acbc0691d3e7e52437"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-24T19:43:56.536400136"
    }
}